dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1350,6 +1350,9 @@ CDefaultContextMenu::InvokePidl(LPCMINVOKECOMMANDINFOEX lpcmi, LPCITEMIDLIST pid
     else if (!StrIsNullOrEmpty(lpcmi->lpParameters) && __SHCloneStrAtoW(&pszParamsW, lpcmi->lpParameters))
         sei.lpParameters = pszParamsW;
 
+    if (!sei.lpClass && (lpcmi->fMask & (CMIC_MASK_HASLINKNAME | CMIC_MASK_HASTITLE)) && unicode)
+        sei.lpClass = lpcmi->lpTitleW; // Forward .lnk path from CShellLink::DoOpen (for consrv STARTF_TITLEISLINKNAME)
+
```

Comment:
Will this fix the following bug?
In WinXP it would apply the new icon to the cmd-windows titlebar (I double-checked that!), but in ros it does not.
[bug_in_unpatched_ros.webm](https://github.com/user-attachments/assets/81d59199-1a25-454c-962e-760876f6352c)

Reply:
> Will this fix the following bug?

Yes